### PR TITLE
How to migrate content attachments easily?

### PR DESCRIPTION
**Pull request by RoySegall**

### CODE REVIEW

**Comment by RoySegall** [comment]
@amitaibu I made a demo module for this PR - https://github.com/RoySegall/comics_migration - can you have a look at this one please? 10x.

**Comment by amitaibu** [comment]
Should comics_migration work on 8.0 or 8.1? (I wasn't able to make it work on 8.1 - so checking what are the expectations).

**Comment by RoySegall** [comment]
wrote it in 8.0 but shold not have any problem in 8.1. I'll check.

**Comment by amitaibu** [comment]
maybe you could create full installation profile, based out of [drupal-porject](https://github.com/drupal-composer/drupal-project) like we do for our closed d8 projects

**Comment by amitaibu** [comment]
What's the status here?

**Comment by RoySegall** [comment]
I started to use the composer project but stopped at one point. I'll try to get back to it.

**Comment by amitaibu** [comment]
yalla, lets get this ready - remember how _you_ looked for it, before finding the solution? ;)

**Comment by RoySegall** [comment]
OK. The demo repo is working and now when running the install file we get site. I'll fine tune the post tomorrow.

**Comment by amitaibu** [comment]
👍

**Comment by RoySegall** [comment]
@amitaibu Ready for another review.

File: app/_posts/2016-06-2-drupal8-migrate-files.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+title: "Drupal 8: migrate with attachments easily"
+tags:
+  - "Drupal 8"
+  - "Drupal-planet"
+  - "Migration"
+permalink: "/content/drupal-8-attachment-migration"
+author: RoySegall
+layout: post
+image: "/assets/images/posts/apps-entity-restrictions/burner.jpg"
```
Comment (amitaibu) [inline app/_posts/2016-06-2-drupal8-migrate-files.md]: wrong thumb

Comment (amitaibu) [inline app/_posts/2016-06-2-drupal8-migrate-files.md]: also an image in the blog post would be nice

**Comment by RoySegall** [comment]
Replaced the image of hipi guy:
gizra_s_blog

**Comment by RoySegall** [comment]
And improved the title.

File: app/_posts/2016-06-2-drupal8-migrate-files.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+---
+title: "Drupal 8: migrate nodes with attachments easily"
+tags:
+  - "Drupal 8"
+  - "Drupal-planet"
+  - "Migration"
+permalink: "/content/drupal-8-attachment-migration"
+author: RoySegall
+layout: post
+image: "/assets/images/posts/drupal-8-migrate/tony.png"
```
Comment (amitaibu) [inline app/_posts/2016-06-2-drupal8-migrate-files.md]: please rename tony to thumb, and follow the [guides](https://github.com/Gizra/gizra.com#technical-stuff-to-write-posts) on how to create it

Comment (amitaibu) [inline app/_posts/2016-06-2-drupal8-migrate-files.md]: also:

1. the thumb should look different from the image (could be like the previous one, where you just showed tony)
1. The image should be inserted in the body of the post as-well

File: app/_posts/2016-06-2-drupal8-migrate-files.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+migration of files/ images out there? (including copy from source into public:// )
+</p>&mdash; Amitai Burstein (@amitaibu) <a href="https://twitter.com/amitaibu/status/718441947325677569">April 8, 2016</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+Few minutes after the tweet was published we received a great hint from the fine folks of
+[Evoloving Web](https://evolvingweb.ca/). They were already migrating files into Drupal 8
+from older Drupal 7, and were kind enough to [blog post](https://evolvingweb.ca/blog/bringing-files-along-for-ride-to-d8) about it.
+
+However, we were still missing another piece of the puzzle, as we wanted to migrate
+files from an outside directory, directly into Drupal. I gave my good friend [@jsacksick](https://twitter.com/jsacksick) a poke (it's easy, as he sits right in front of me), and he gave me the answer in a silver plate.
```
Comment (IshaDakota) [inline app/_posts/2016-06-2-drupal8-migrate-files.md]: "on a silver platter"

File: app/_posts/2016-06-2-drupal8-migrate-files.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  title: _title
+  field_image:
+    source: _image
+    plugin: file_import
+  field_alter_ego: _alter_ego
+  'body/value': _description
+```
+
+In Drupal 7, when we wanted to prepare the value before
+populating the entity fields as we want, we changed it in a
```
Comment (IshaDakota) [inline app/_posts/2016-06-2-drupal8-migrate-files.md]: "as we wanted"

File: app/_posts/2016-06-2-drupal8-migrate-files.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+In our case, the `transform` method is responsibly for registrating the new file into Drupal
+using `file_unmanaged_copy` and friends.
+
+## Conclusion
+
+Some of the know hows and best practices are still missing from Drupal 8. But they
+can and should be re-learned and re-published. So remember kids, if you ever feel
+frustrated about not finding a solution, always
+reach out to smart community members (and then write a post about it, so everybody ).
```
Comment (IshaDakota) [inline app/_posts/2016-06-2-drupal8-migrate-files.md]: so everybody... ?

File: app/_posts/2016-06-2-drupal8-migrate-files.md (hunk 2 above)
Comment (amitaibu) [inline app/_posts/2016-06-2-drupal8-migrate-files.md]: @RoySegall ^^

File: app/_posts/2016-06-2-drupal8-migrate-files.md (hunk 4 above)
Comment (RoySegall) [inline app/_posts/2016-06-2-drupal8-migrate-files.md]: Actually - `when we needed`.

File: app/_posts/2016-06-2-drupal8-migrate-files.md (hunk 5 above)
Comment (RoySegall) [inline app/_posts/2016-06-2-drupal8-migrate-files.md]: Maybe he run out of juice at the last words?

I think the missing words are "could enjoy it."

**Comment by RoySegall** [comment]
@amitaibu Added some photos:

drupal_8__migrate_nodes_with_attachments_easily_
gizra_s_blog_

**Comment by amitaibu** [comment]
@IshaDakota can you please review

**Comment by IshaDakota** [comment]
Made some minor grammar/typo edits.

Two notes:

1. Make sure I got this meaning correct:

> In Drupal 7, because we wanted to prepare the value before populating the entity fields, we did it in a prepare method. In Drupal 8 we have process plugins.

2. For future posts, because titles are no longer all caps, title capitalization matters.

**Comment by amitaibu** [comment]
deployed, thanks,

**Pull request merged**